test/functional/helpers.lua

```diff
@@ -518,8 +519,29 @@ function module.set_shell_powershell()
     or  {'alias:echo'}, ',')..';'
   module.source([[
     let &shell = ']]..shell..[['
-    set shellquote= shellpipe=\| shellxquote=
-    let &shellredir = '| Out-File -Encoding UTF8'
+    set shellquote= shellxquote=
+    let &shellpipe = '2>&1 | Out-File -Encoding UTF8 %s; exit $LastExitCode'
+    let &shellredir = '2>&1 | Out-File -Encoding UTF8 %s; exit $LastExitCode'
+    let &shellcmdflag = '-NoLogo -NoProfile -ExecutionPolicy RemoteSigned -Command ]]..cmd..[['
+  ]])
+end
+
+-- PowerShell Core
+function module.has_pwsh()
+  return module.eval('executable("pwsh")') == 1
+end
+
+function module.set_shell_pwsh()
+  -- NOTE This config works with current PowerShell Core 7.1.3
+  assert(module.has_pwsh())
+  local cmd = 'Remove-Item -Force '..table.concat(iswin()
+    and {'alias:cat', 'alias:echo', 'alias:sleep'}
+    or  {'alias:echo'}, ',')..';'
+  module.source([[
+    let &shell = 'pwsh'
+    set shellquote= shellxquote=
+    let &shellredir='2>&1 | Out-File %s; exit $LastExitCode'
```

Comment:
Always UTF8, even on Windows?

Reply:
According to [this doc](https://docs.microsoft.com/en-us/powershell/scripting/dev-cross-plat/vscode/understanding-file-encoding?view=powershell-7.1#configuring-powershell), yes. 

> In PowerShell 6+, the default encoding is UTF-8 without BOM on all platforms.

Reply:
But naturally, what's actually happening is not following the docs 😑... Back to the drawing board
```
PowerShell 7.1.3
Copyright (c) Microsoft Corporation.

https://aka.ms/powershell
Type 'help' to get help.

PS C:\Users\dtcaciuc> [Console]::OutputEncoding

IsSingleByte      : True
EncodingName      : OEM United States
WebName           : ibm437
HeaderName        : ibm437
BodyName          : ibm437
```